sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue

```diff
@@ -179,7 +136,29 @@ export default {
         }
       }
     };
-
+    const submitCOEDeny = async (
+      submissionData: DenyConfirmationOfEnrollment,
+    ) => {
+      try {
+        await ConfirmationOfEnrollmentService.shared.denyConfirmationOfEnrollment(
+          props.locationId,
+          props.applicationId,
+          submissionData,
+        );
+        toast.success("COE is Denied", "Application Status Has Been Updated.");
+        router.push({
+          name: InstitutionRoutesConst.COE_SUMMARY,
+        });
+      } catch {
+        toast.error(
+          "Unexpected error",
+          "An error happened while denying Confirmation of Enrollment.",
+        );
+      }
+    };
+    const denyProgramInformation = async () => {
+      await denyCOEModal.value.showModal();
```

Comment:
IMO, if the modal is already awaiting a response, what is the benefit of registering/triggering an event?
All the event-related code could be removed and the feature would still be working but with less code.
If the idea is to keep the logic for submitting in a separate method, it could be done checking the response from the showDialog and calling _submitCOEDeny_ after line 160.